### PR TITLE
README: fix style issues/add flash instructions

### PR DESCRIPTION
**Pull request by igoropaniuk**
1. Address markdown style issues reported by the linter
2. Add initial flash instructions for Qualcomm® RB3 Gen 2 Development Kit.

### CODE REVIEW

**Review by mbolivar — APPROVED** [review]
LGTM, thanks!

File: README.md [hunk 1]
```diff
@@ -62,6 +62,80 @@ an easy way to setup bitbake based projects. For more details, visit the
 
 For a manual build without KAS, refer to the [Yocto Project Quick Build](https://docs.yoctoproject.org/brief-yoctoprojectqs/index.html).
 
+## Flash
+
+### Build QDL tool
+
+QDL tool communicates with USB devices of id `05c6:9008` and uploads a flash
```
Comment (mwasilew) [inline README.md:69]: change "id" to "VID:PID"? qdl can work with multiple devices connected to the same host using the device unique ID. This is done by adding flag "--serial"

File: README.md [hunk 2]
```diff
@@ -62,6 +62,80 @@ an easy way to setup bitbake based projects. For more details, visit the
...
+
+   ```
+   cd build/tmp/deploy/images/qcs6490-rb3gen2-core-kit/
+   tar zxvf core-image-base-qcs6490-rb3gen2-core-kit.rootfs.qcomflash.tar.gz
+   ```
+
+6. Use the QDL tool (built in the previous section) to flash the images:
+
+   ```
+   qdl --debug prog_firehose_ddr.elf rawprogram*.xml patch*.xml
```
Comment (mwasilew) [inline README.md:127]: add "--serial" ? Serial can be obtained using `lsusb -v -d 05c6:9008`

File: README.md [hunk 3]
```diff
@@ -62,6 +62,80 @@ an easy way to setup bitbake based projects. For more details, visit the
...
+   ```
+   cd qdl
+   make
+   ```
+
+### Flash images
+
+Make sure that ModemManager is not running, disable it if necessary.
+Location of all DIP switches, USB debug port and buttons (`F_DL` for instance)
+can be found on [RB3 Gen 2 Quick Start Guide](https://docs.qualcomm.com/bundle/publicresource/topics/80-70014-253/ubuntu_host.html).
```
Comment (ricardosalveti) [inline README.md:97]: I think having the generic instructions here is fine, but this layer is expected to support multiple machines (there is already support for ride), so should we isolate the board specific instructions and just have a link instead?

**Review by mwasilew — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (mwasilew) [reply README.md:97]: For RIDE one needs to use alpaca to set the EDL mode. I don't think there are public instructions available for that.

Comment (igoropaniuk) [reply README.md:97]: @ricardosalveti makes sense, will do

File: README.md (hunk 1 above)
Comment (igoropaniuk) [reply README.md:69]: That's from QDL tool readme https://github.com/linux-msm/qdl/blob/master/README, based on that I assumed that all Q devices use the same `VID:PID` in EDL mode

File: README.md [hunk 4]
```diff
@@ -23,47 +25,135 @@ revision: HEAD
...
+
+1. Set up `DIP_SW_0` positions `1` and `2` to `ON`. This enables serial output
+   to the debug port.
+2. Connect the mini USB debug cable to the host. Baud rate should be `115200`.
```
Comment (igoropaniuk) [inline README.md:101]: should be `s/mini/micro/g`

File: README.md (hunk 1 above)
Comment (mwasilew) [reply README.md:69]: That's correct, but if you have multiple devices connected to the same host, they will have different serial ID: https://github.com/linux-msm/qdl/blob/master/qdl.c#L131

File: README.md (hunk 2 above)
Comment (igoropaniuk) [reply README.md:127]: will do, thanks

File: README.md (hunk 1 above)
Comment (igoropaniuk) [reply README.md:69]: ah, ok, fill fix

File: README.md (hunk 3 above)
Comment (igoropaniuk) [reply README.md:102]: I've tried to make flash instructions as much board-agnostic as possible. I've moved board specific details to a dedicated section.

File: README.md (hunk 1 above)
Comment (igoropaniuk) [reply README.md]: fixed

File: README.md (hunk 4 above)
Comment (igoropaniuk) [reply README.md]: fixed

File: README.md (hunk 2 above)
Comment (igoropaniuk) [reply README.md:139]: added information about `--serial` param

**Comment by igoropaniuk** [comment]
@ricardosalveti @mwasilew any additional comments/suggestions?

**Comment by igoropaniuk** [comment]
@ricardosalveti @mwasilew @ndechesne  any additional comments/suggestions?

**Comment by ndechesne** [comment]
I am not sure I like duplicating the documentation here... but for now we should go ahead and merge. we can always reorganize things later once we have a better idea about how the whole thing will work out.

**Review by ndechesne — APPROVED** [review]

**Comment by igoropaniuk** [comment]
Rebased onto the latest main

**Review by ricardosalveti — APPROVED** [review]
LGTM

**Comment by ricardosalveti** [comment]
In order to merge we need a +1 from @mwasilew, as he requested changes.

**Review by mwasilew — APPROVED** [review]

**Comment by mwasilew** [comment]
Sorry, I missed the fixes from @igoropaniuk.

**Pull request merged**